protocol/native/native.go

```diff
@@ -36,7 +38,28 @@ func ExecParser(toNoder ToNoder, path string, args ...string) (*Parser, error) {
 }
 
 func (p *Parser) ParseUAST(req *protocol.ParseUASTRequest) *protocol.ParseUASTResponse {
-	nativeResp, err := p.Client.ParseNativeAST(&ParseASTRequest{Content: req.Content})
+	// If the request encoding is Base64 try to decode it as UTF8. Leave it
+	// as Base64 if it isn't possible.
```

Comment:
From the other point of view: if we can't decode it to UTF8 without problems at that point, why not doing it before passing the code to the native driver? It's not that the UTF8 encoding is going to be alias to the driver - it's the one that the JSON format supports so it should be able to handle it.